### PR TITLE
Fix spark dev errors

### PR DESCRIPTION
**Pull request by harupy**
[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/9800?quickstart=1)

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Fix:

https://github.com/mlflow-automation/mlflow/actions/runs/6380638079/job/17323631710#step:13:275

```
                    proc = Popen(command, **popen_kwargs)
    
                # Wait for the file to appear, or for the process to exit, whichever happens first.
                while not proc.poll() and not os.path.isfile(conn_info_file):
                    time.sleep(0.1)
    
                if not os.path.isfile(conn_info_file):
>                   raise PySparkRuntimeError(
                        error_class="JAVA_GATEWAY_EXITED",
                        message_parameters={},
                    )
E                   pyspark.errors.exceptions.base.PySparkRuntimeError: [JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
```

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/ml-package-versions.yml [hunk 1]
```diff
@@ -311,7 +311,7 @@ spark:
       # -Pconnect -Phive is requried by spark connect
       ./build/mvn -Pconnect -Phive -DskipTests --no-transfer-progress clean package
       cd python
-      python setup.py install
+      pip install '.[connect]'
```
Comment (WeichenXu123) [inline mlflow/ml-package-versions.yml:314]: good finding! :) let's see CI result

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Comment by harupy** [comment]
@WeichenXu123 A new error occurred:

```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
E                   : java.lang.NoSuchMethodError: 'org.sparkproject.connect.grpc.ServerServiceDefinition org.apache.spark.connect.proto.SparkConnectServiceGrpc.bindService(org.apache.spark.connect.proto.SparkConnectServiceGrpc$AsyncService)'
E                   	at org.apache.spark.sql.connect.service.SparkConnectService.bindService(SparkConnectService.scala:242)
E                   	at org.sparkproject.connect.grpc.internal.ServerImplBuilder.addService(ServerImplBuilder.java:143)
E                   	at org.sparkproject.connect.grpc.internal.ServerImplBuilder.addService(ServerImplBuilder.java:57)
E                   	at org.sparkproject.connect.grpc.internal.AbstractServerImplBuilder.addService(AbstractServerImplBuilder.java:93)
E                   	at org.sparkproject.connect.grpc.internal.AbstractServerImplBuilder.addService(AbstractServerImplBuilder.java:48)
E                   	at org.apache.spark.sql.connect.service.SparkConnectService$.startGRPCService(SparkConnectService.scala:417)
E                   	at org.apache.spark.sql.connect.service.SparkConnectService$.start(SparkConnectService.scala:433)
E                   	at org.apache.spark.sql.connect.SparkConnectPlugin$$anon$1.init(SparkConnectPlugin.scala:48)
E                   	at org.apache.spark.internal.plugin.DriverPluginContainer.$anonfun$driverPlugins$1(PluginContainer.scala:53)
E                   	at scala.collection.StrictOptimizedIterableOps.flatMap(StrictOptimizedIterableOps.scala:118)
E                   	at scala.collection.StrictOptimizedIterableOps.flatMap$(StrictOptimizedIterableOps.scala:105)
E                   	at scala.collection.immutable.ArraySeq.flatMap(ArraySeq.scala:35)
E                   	at org.apache.spark.internal.plugin.DriverPluginContainer.<init>(PluginContainer.scala:46)
E                   	at org.apache.spark.internal.plugin.PluginContainer$.apply(PluginContainer.scala:210)
E                   	at org.apache.spark.internal.plugin.PluginContainer$.apply(PluginContainer.scala:193)
E                   	at org.apache.spark.SparkContext.<init>(SparkContext.scala:[578](https://github.com/mlflow/mlflow/actions/runs/6389947885/job/17342237962?pr=9800#step:13:579))
E                   	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
E                   	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
E                   	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
E                   	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
E                   	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
E                   	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
E                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
E                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
E                   	at py4j.Gateway.invoke(Gateway.java:238)
E                   	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
E                   	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
E                   	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
E                   	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
E                   	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by harupy** [comment]
The original error has been fixed. I'll merge this PR.

**Pull request merged**